cs/test/core/GuidConversionTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace UnitTest
+{
+    using System;
+    using Bond;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Summary description for GuidConversionTests
+    /// </summary>
+    [TestFixture]
+    public class GuidConversionTests
+    {
+        [Test]
+        public void ImplicitConversion()
+        {
+            var systemGuid = Guid.NewGuid();
+            var bondGuid = (GUID)systemGuid;
+
+            // The Datas should be populated if it converted correctly
+            Assert.AreNotEqual(bondGuid.Data1, 0);

```

Comment:
Tried to use your IPAddress trick in the conversion. Could not since it is not available in the portable .NET library used int he project.
Instead used BitConverter.IsLittleEndian and had two separate paths. Hopefully it's not too horrible